app/services/users/free_balance.rb

```diff
@@ -0,0 +1,22 @@
+class Users::FreeBalance < Struct.new(:user)
+  def call
+    free_balance
+  end
+
+  private
+
+  def free_balance
+    balance - transactions_amount
+  end
+
+  def transactions_amount
+    category_transactions.sum(&:amount)
+  end
+
+  def category_transactions
+    user_transactions.category_transactions
```

Comment:
@marderer can't we get `category_transactions ` from `user`? just `user.category_transactions`
If we can't then let's add such ability